db/migrate/20240424160513_add_hide_attachments_to_projects.rb

```diff
@@ -26,17 +26,10 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
 # See COPYRIGHT and LICENSE files for more details.
-# ++
+#++
 
-module Admin::QuarantinedAttachments
-  # rubocop:disable OpenProject/AddPreviewForViewComponent
-  class IndexPageHeaderComponent < ApplicationComponent
-    include ApplicationHelper
-
-    def breadcrumb_items
-      [{ href: admin_index_path, text: t("label_administration") },
-       { href: admin_settings_attachments_path, text: t("attributes.attachments") },
-       t("antivirus_scan.quarantined_attachments.title")]
-    end
+class AddHideAttachmentsToProjects < ActiveRecord::Migration[7.1]
+  def change
+    add_column :projects, :hide_attachments, :boolean, default: false
```

Comment:
🟡 So, we do not have something like a settings table for projects? Feels odd to put this boolean into the project model. What were your thoughts, @ulferts ?

Reply:
Yes, it seem like we do not have project settings table or something similar. In the current database structure projects table seems to be right place for this boolean. I do not see it as a problem at all. If you have a better place for it feel free to suggest it.

Reply:
I talked a lot with Wieland, Oliver and Jens, and we agreed to use a `settings` column, which is of type jsonb. Accessible with `store_attribute`.

I also updated the behavior of the `work_package.hide_attachments?` method. Only fall back to global setting, if setting is unset.